src/EventSubscriber/WebSSOEventSubscriber.php

```diff
@@ -0,0 +1,312 @@
+     *
+     * @param WebSSOConfiguration $webSSOConfiguration
+     * @throws \InvalidArgumentException
+     */
+    private function validateLoginAttributeOrFail(WebSSOConfiguration $webSSOConfiguration): void
+    {
+        if (! array_key_exists($webSSOConfiguration->getLoginHeaderAttribute(), $_SERVER)) {
+            throw new \InvalidArgumentException('Missing Login Attribute');
+        }
+    }
+
+    /**
+     * Find User or throw an exception
+     *
+     * @param string $alias
+     * @return Contact
+     * @throws NotFoundException
+     */
+    private function findUserByAliasOrFail(string $alias): Contact
+    {
+        $user = $this->contactRepository->findByName($alias);
+        if ($user === null) {
+            throw new NotFoundException("Contact $alias doesn't exists");
+        }
+
+        return $user;
+    }
+
+    /**
+     * Create the session
+     *
+     * @param Contact $user
+     * @param Request $request
+     */
+    private function createSession(Contact $user, Request $request): void
+    {
+        global $pearDB;
+        $pearDB = $this->dependencyInjector['configuration_db'];
+        $sessionUserInfos = [
+            'contact_id' => $user->getId(),
+            'contact_name' => $user->getName(),
+            'contact_alias' => $user->getAlias(),
+            'contact_email' => $user->getEmail(),
+            'contact_lang' => $user->getLang(),
+            'contact_passwd' => $user->getEncodedPassword(),
+            'contact_autologin_key' => '',
+            'contact_admin' => $user->isAdmin() ? '1' : '0',
+            'default_page' => $user->getDefaultPage(),
+            'contact_location' => $user->getLocale(),
+            'show_deprecated_pages' => $user->isUsingDeprecatedPages(),
+            'reach_api' => $user->hasAccessToApiConfiguration() ? 1 : 0,
+            'reach_api_rt' => $user->hasAccessToApiRealTime() ? 1 : 0
+        ];
+        $centreonSession = new \Centreon($sessionUserInfos);
+        $request->getSession()->start();
+        $request->getSession()->set('centreon', $centreonSession);
+        $_SESSION['centreon'] = $centreonSession;
+    }
+
+    /**
+     * Create token if not exist
+     *
+     * @param string $sessionId
+     * @param integer $webSSOConfigurationId
+     * @param Contact $user
+     * @param string $clientIp
+     */
+    private function createTokenIfNotExist(
+        string $sessionId,
+        int $webSSOConfigurationId,
+        Contact $user,
+        string $clientIp
+    ): void {
+        $authenticationTokens = $this->authenticationService->findAuthenticationTokensByToken(
+            $sessionId
+        );
+        if ($authenticationTokens === null) {
+            $sessionExpireOption = $this->optionService->findSelectedOptions(['session_expire']);
+            if (!empty($sessionExpireOption)) {
+                $this->sessionExpirationDelay = (int) $sessionExpireOption[0]->getValue();
```

Comment:
`sessionExpirationDelay` seems already injected by symfony
maybe this should be removed from dependency injection, since it's a value stored in database

Reply:
yup, we did the same mechanism in the local provider but as i see, the session_expire is a mandatory parameter in Centreon UI general config form, so it could be removed